packages/opentelemetry-types/src/trace/timed_event.ts

```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Event } from './event';
+
+/**
+ * Represents a timed event.
+ * A timed event is an event with a timestamp.
+ */
+export interface TimedEvent {
```

Comment:
What's the motivation for making Event and TimedEvent distinct? Is that specified in the spec?

Reply:
Not explicitly. Here's [the spec that mentions them ](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/tracing-api.md#gettimedevents)and here is the [Java implementation](https://github.com/open-telemetry/opentelemetry-java/blob/master/api/src/main/java/io/opentelemetry/trace/SpanData.java#L265).

A timed event has a timestamp attached to it, so to me, it makes sense to create a type for it and I think that was also the reason why Java does it like that.

Reply:
I don't think there needs to be a distinction. An event always has a time. This is a fact not only in software but also in the physical world.

Reply:
`Event` does not have a time and there is a `TimedEvent` in the spec. I think that we should stick to the spec and discuss and deviation from it with the spec team.

Reply:
There are so many things wrong with the spec that I don't think this is the best approach.

Reply:
Yeah I am confused why there would be an event with no time, seems like a leftover from opencensus's implementation that ended up in the API.